                                   Machine Learning for Data Sciences
                                           Lab assessment 1
Name     :Sheema zainab M I    

Reg no   :19MID0014

Objective:Perform logistic regression on data set from UCI repository

Dataset  :Breast Cancer Detection ( wisconsin) diagnosis

Link to data set :  https://archive-beta.ics.uci.edu/ml/datasets/breast+cancer+wisconsin+diagnostic 

Goal  : To classify is a breast cancer tumor is malignant or benign

In [1]:
print("SHEEMA ZAINAB M I 19MID0014")

SHEEMA ZAINAB M I 19MID0014


# IMPORTING LIBRARIES AND DATA SET

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [3]:
df=pd.read_csv("uci_cancer.csv")
df.head()

id diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
0    842302         M        17.99         10.38          122.80     1001.0   
1    842517         M        20.57         17.77          132.90     1326.0   
2  84300903         M        19.69         21.25          130.00     1203.0   
3  84348301         M        11.42         20.38           77.58      386.1   
4  84358402         M        20.29         14.34          135.10     1297.0   

   smoothness_mean  compactness_mean  concavity_mean  concave points_mean  \
0          0.11840           0.27760          0.3001              0.14710   
1          0.08474           0.07864          0.0869              0.07017   
2          0.10960           0.15990          0.1974              0.12790   
3          0.14250           0.28390          0.2414              0.10520   
4          0.10030           0.13280          0.1980              0.10430   

   ...  texture_worst  perimeter_worst  area_worst  smoothness_worst  \
0  ...          17.33           184.60      2019.0            0.1622   
1  ...          23.41           158.80      1956.0            0.1238   
2  ...          25.53           152.50      1709.0            0.1444   
3  ...          26.50            98.87       567.7            0.2098   
4  ...          16.67           152.20      1575.0            0.1374   

   compactness_worst  concavity_worst  concave points_worst  symmetry_worst  \
0             0.6656           0.7119                0.2654          0.4601   
1             0.1866           0.2416                0.1860          0.2750   
2             0.4245           0.4504                0.2430          0.3613   
3             0.8663           0.6869                0.2575          0.6638   
4             0.2050           0.4000                0.1625          0.2364   

   fractal_dimension_worst  Unnamed: 32  
0                  0.11890          NaN  
1                  0.08902          NaN  
2                  0.08758          NaN  
3                  0.17300          NaN  
4                  0.07678          NaN  

[5 rows x 33 columns]

# DROPPING UNWANTED ATTRIBUTES

In [3]:

df = df.drop('Unnamed: 32', axis=1)

# ASSIGNING INPUT DATA IN X MATRIX AND OUTPUT VALUE IN Y VECTOR

In [5]:
X=df.iloc[:,2:32].values
y=df.diagnosis


# ENCODING CATEGORICAL DATA

In [6]:
#Encoding categorical data values
from sklearn.preprocessing import LabelEncoder
labelencoder_y = LabelEncoder()
y = labelencoder_y.fit_transform(y)

# SPLITTING DS INTO TRAIN AND TEST

In [7]:
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, y, test_size = 0.30, random_state = 42)

# FEATURE SCALING 

In [8]:

from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

# APPLYING LOGISTIC REGRESSION MODEL 

In [9]:
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression(random_state = 42)
classifier.fit(X_train, Y_train)

C:\Users\sheem\Anaconda3\lib\site-packages\sklearn\linear_model\least_angle.py:35: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  eps=np.finfo(np.float).eps,
C:\Users\sheem\Anaconda3\lib\site-packages\sklearn\linear_model\least_angle.py:597: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  eps=np.finfo(np.float).eps, copy_X=True, fit_path=Tru

LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
          intercept_scaling=1, max_iter=100, multi_class='warn',
          n_jobs=None, penalty='l2', random_state=42, solver='warn',
          tol=0.0001, verbose=0, warm_start=False)

In [10]:
y_pred = classifier.predict(X_test)

C:\Users\sheem\Anaconda3\lib\site-packages\sklearn\linear_model\base.py:283: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  indices = (scores > 0).astype(np.int)


# CONFUSION MATRIX

In [11]:

from sklearn.metrics import confusion_matrix
cm = confusion_matrix(Y_test, y_pred)
cm

array([[106,   2],
       [  1,  62]], dtype=int64)

In [12]:
from sklearn import metrics


In [13]:
TP = cm[1, 1]
TN = cm[0, 0]
FP = cm[0, 1]
FN = cm[1, 0]

# ACCURACY SCORE

In [16]:
print('Accuracy Score on confusion_matrix: ')
print((TP + TN) / float(TP + TN + FP + FN))

Accuracy Score on confusion_matrix: 
0.9824561403508771


# ERROR SCORE


In [17]:
print('Error Score on confusion_matrix: ' ,1 - metrics.accuracy_score(Y_test, y_pred))

Error Score on confusion_matrix:  0.01754385964912286


# SENSITIVITY/RECALL

In [14]:
sensitivity = TP / float(FN + TP)
print('sensitivity Score on confusion_matrix: ',metrics.recall_score(Y_test, y_pred))

sensitivity Score on confusion_matrix:  0.9841269841269841


# PRECISION

In [15]:

precision = TP / float(TP + FP)
print('precision Score on confusion_matrix: ',metrics.precision_score(Y_test, y_pred))

precision Score on confusion_matrix:  0.96875


# F1 SCORE

In [23]:
f1=2*(precision*sensitivity)/(precision+sensitivity)
f1

0.9763779527559054

# CONCLUSION

We conclude that the  

Accuracy = 98.245 %

Error rate = 1.754% 

Precision =  96.87% 

Recall = 98.41% 

f1 Score= 97.63%

We can say from these observations that our model will perform well .